flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/ResourceManagerGateway.java

```diff
@@ -266,4 +267,12 @@ void heartbeatFromTaskManager(
      */
     CompletableFuture<ThreadDumpInfo> requestThreadDump(
             ResourceID taskManagerId, @RpcTimeout Time timeout);
+
+    /**
+     * Requests the {@link TaskExecutorGateway}.
+     *
+     * @param taskManagerId identifying the {@link TaskExecutor}.
+     * @return Future containing the task executor gateway.
+     */
+    CompletableFuture<TaskExecutorGateway> requestTaskExecutorGateway(ResourceID taskManagerId);
```

Comment:
We generally don't expose other component gateways, because it make it unnecessarily difficult to mock the RMGateway in tests.

Reply:
Flame Graphs component needs to be able to send requests to the `TaskExecutors` that are responsible for running the subtasks of the vertex selected in the UI. What would be the alternative?
P.S.: it has been an explicit request from Till to avoid adding non-core functionality to the `JobManagerSharedServices`, the way the old backpressure sampler was implemented, and rather rely on the `ArchivedExecutionGraph` which is already available to the `WebMonitorEndpoint`.

Reply:
Now returning only the extracted `TaskExecutorThreadInfoGateway`.